language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/AddModuleToBallerinaTomlCodeAction.java

```diff
@@ -0,0 +1,168 @@
+                versions);
+        if (versions.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Optional<BallerinaToml> toml = project.get().currentPackage().ballerinaToml();
+        if (toml.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Position dependencyStart = new Position(getDependencyStartLine(toml.get()), 0);
+        
+        String dependency = "[[dependency]]\n" +
+                "org = \"" + orgName + "\"\n" +
+                "name = \"" + pkgName + "\"\n" +
+                "version = \"" + getLatestVersion(versions) + "\"\n" +
+                "repository = \"local\"\n\n";
+        TextEdit textEdit = new TextEdit(new Range(dependencyStart, dependencyStart), dependency);
+        CodeAction action = CodeActionUtil.createCodeAction(commandTitle, List.of(textEdit), 
+                project.get().sourceRoot().resolve(ProjectConstants.BALLERINA_TOML).toString(), 
+                CodeActionKind.QuickFix);
+        return Collections.singletonList(action);
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+    
+    private void getVersions(LSPackageLoader lsPackageLoader, Project project, String orgName, String pkgName, 
+                             ArrayList<PackageVersion> versions) {
+        BallerinaUserHome ballerinaUserHome = BallerinaUserHome.from(project.projectEnvironmentContext().environment());
+        PackageRepository localRepository = ballerinaUserHome.localPackageRepository();
+        List<ModuleInfo> modules = lsPackageLoader.getLocalRepoPackages(localRepository);
+        for (ModuleInfo mod : modules) {
+            if (mod.packageOrg().value().equals(orgName) && mod.packageName().value().equals(pkgName)) {
+                versions.add(mod.packageVersion());
+            }
+        }
+    }
+    
+    private String getLatestVersion(ArrayList<PackageVersion> versions) {
+        PackageVersion latestVersion = versions.get(0);
+        for (int i = 1; i < versions.size(); i++) {
+            PackageVersion version = versions.get(i);
+            if (version.compareTo(latestVersion) == SemanticVersion.VersionCompatibilityResult.GREATER_THAN) {
+                latestVersion = version;
+            }
+        }
+        return latestVersion.toString();
+    }
+    
+    private int getDependencyStartLine(BallerinaToml toml) {
+        DocumentNode tomlSyntaxTree = toml.tomlDocument().syntaxTree().rootNode();
+        NodeList<DocumentMemberDeclarationNode> members = tomlSyntaxTree.members();
+        for (DocumentMemberDeclarationNode member : members) {
+            if (member.kind() == SyntaxKind.TABLE) {
+                TableNode tableNode = (TableNode) member;
+                if (TomlSyntaxTreeUtil.toQualifiedName(tableNode.identifier().value()).equals("package")) {
+                    return tableNode.lineRange().endLine().line() + 2;
+                }
+            }
+        }   
+        return 0;
```

Comment:
```suggestion
tomlSyntaxTree.members().stream()
                .filter(member-> member.kind() == io.ballerina.toml.syntax.tree.SyntaxKind.TABLE
                        && ((TableNode)member).identifier().value().equals("package"))
                .findFirst().map(tableNode-> tableNode.lineRange().endLine().line() + 2).orElse(0);
```

Reply:
This is just replacing using stream, right?

Reply:
Yes but streams can perform better in terms of memory. Anyway, we can keep the original  as this will not have a significant impact. Ignore the comment.